components/fxa-client/examples/auth_code_session.rs

```diff
@@ -0,0 +1,200 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use cli_support::prompt::prompt_string;
+use dialoguer::Select;
+use fxa_client::{auth, device, Config, FirefoxAccount, IncomingDeviceCommand};
+use serde_json::json;
+use std::{
+    collections::HashMap,
+    fs,
+    io::{Read, Write},
+    sync::{Arc, Mutex},
+    thread, time,
+};
+use url::Url;
+use viaduct::Request;
+
+static CREDENTIALS_PATH: &str = "credentials.json";
+static CONTENT_SERVER: &str = "https://stable.dev.lcip.org";
+static CLIENT_ID: &str = "3c49430b43dfba77";
+static REDIRECT_URI: &str = "https://stable.dev.lcip.org/oauth/success/3c49430b43dfba77";
+static SCOPES: &[&str] = &["profile", "https://identity.mozilla.com/apps/oldsync"];
+static DEFAULT_DEVICE_NAME: &str = "Bobo device";
+
+use anyhow::Result;
+
+fn load_fxa_creds() -> Result<FirefoxAccount> {
+    let mut file = fs::File::open(CREDENTIALS_PATH)?;
+    let mut s = String::new();
+    file.read_to_string(&mut s)?;
+    Ok(FirefoxAccount::from_json(&s)?)
+}
+
+fn load_or_create_fxa_creds(cfg: Config) -> Result<FirefoxAccount> {
+    let acct = load_fxa_creds().or_else(|_e| create_fxa_creds(cfg))?;
+    persist_fxa_state(&acct);
+    Ok(acct)
+}
+
+fn persist_fxa_state(acct: &FirefoxAccount) {
+    let json = acct.to_json().unwrap();
+    let mut file = fs::OpenOptions::new()
+        .read(true)
+        .write(true)
+        .truncate(true)
+        .create(true)
+        .open(CREDENTIALS_PATH)
+        .unwrap();
+    write!(file, "{}", json).unwrap();
+    file.flush().unwrap();
+}
+
+fn create_fxa_creds(cfg: Config) -> Result<FirefoxAccount> {
+    let mut acct = FirefoxAccount::with_config(cfg.clone());
```

Comment:
To make this example work the way the flow would work in the real world, I suggest making two separate `FirefoxAccount` instances and pretend that you're doing a pairing flow between them. The first one would be the "authority" side of the pairing flow, and you'd sign in to it manually using the "enter password" logic that you have here now.  The second instance would be the "supplicant" side of the pairing flow, and you would:

* Call `supplicant.begin_auth_flow()` to obtain an OAuth request URL.
* Parse the OAuth parameters out of the returned OAuth URL (`client_id`, `scope`, `code_challenge`, etc)
* Call `authority.authorize_code_using_session_token()` passing it all the OAuth params that you parsed out above. This will return an OAuth code.
* Use the returned OAuth code to call `supplicant.complete_oauth_flow()` to complete the login process.

In this way, you've used an alreadys-signed-in FirefoxAccount instance to transfer the signed-in state to a new instance, much like we'd want to do with two separate apps in the wild.